client/src/debug/FabricDebugConfigurationProvider.ts

```diff
@@ -18,102 +18,144 @@ import { FabricRuntimeManager } from '../fabric/FabricRuntimeManager';
-        }
+            if (!config.request) {
+                config.request = 'launch';
+            }
 
-        if (!config.args.includes('start')) {
-            config.args.push('start');
-        }
+            if (!config.program) {
+                config.program = path.join(folder.uri.fsPath, 'node_modules', '.bin', 'fabric-chaincode-node');
+            }
 
-        if (!config.args.includes('--peer.address')) {
-            const peerAddress: string = await runtime.getChaincodeAddress();
-            config.args.push('--peer.address', peerAddress);
-        }
+            const tsFiles: Array<vscode.Uri> = await vscode.workspace.findFiles(new vscode.RelativePattern(folder, '**/*.ts'), '**/node_modules/**', 1);
+
+            if (tsFiles.length > 0) {
+                let dir: string = '';
+                const tsConfig: any = await this.loadJSON(folder, 'tsconfig.json');
+                if (tsConfig && tsConfig.compilerOptions && tsConfig.compilerOptions.outDir) {
+                    const outDir: string = tsConfig.compilerOptions.outDir;
+                    if (!path.isAbsolute(outDir)) {
+                        dir = outDir;
+                        if (dir.indexOf('./') === 0) {
+                            dir = dir.substr(2);
+                        }
+                    }
+                    config.preLaunchTask = 'tsc: build - tsconfig.json';
+                }
 
-        if (!config.env) {
-            config.env = {};
-        }
+                const outFilesPath: string = path.join(folder.uri.fsPath, dir, '**/*.js');
 
-        if (!config.env.CORE_CHAINCODE_ID_NAME) {
-            const info: { name: string, version: string } = await this.getContractNameAndVersion(folder);
-            config.env.CORE_CHAINCODE_ID_NAME = `${info.name}:${info.version}`;
-        }
+                if (config.outFiles) {
+                    config.outFiles.push(outFilesPath);
+                } else {
+                    config.outFiles = [outFilesPath];
+                }
+            }
+
+            if (!config.cwd) {
+                config.cwd = folder.uri.fsPath;
+            }
+
+            if (!config.args) {
+                config.args = [];
+            }
 
-        return config;
+            if (!config.args.includes('start')) {
+                config.args.push('start');
+            }
+
+            if (!config.args.includes('--peer.address')) {
+                const peerAddress: string = await this.runtime.getChaincodeAddress();
+                config.args.push('--peer.address', peerAddress);
+            }
+
+            return config;
+        } catch (error) {
+            console.log(error);
+        }
     }
 
     // tslint:disable-next-line no-empty
     public dispose(): void {
     }
 
+    private getNewVersion(): string {
+        const date: Date = new Date();
+        return `${this.PREFIX}-${date.getHours()}${date.getMinutes()}${date.getDate()}${date.getMonth()}${date.getFullYear()}`;
```

Comment:
This should be YYYYMMDDHHMM so it can be easily sorted in chronological order 😉

Reply:
Also, for consistency suggest you ensure the number of digits (12) for each field.